OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -304,8 +330,14 @@ public void ResolveOrder(Actor self, Order order)
 
 					break;
 				case "PauseProduction":
-					if (queue.Count > 0 && queue[0].Item == order.TargetString)
-						queue[0].Pause(order.ExtraData != 0);
+					foreach (var item in queue)
```

Comment:
instead of using the foreach loop here you could use LinQ, aka queue.firstOrDefault((item) => { return item.Item == order.TargetString})?.Pause(order.ExtraData != 0) or something similar

Reply:
?. interesting, thanks!

Reply:
?. makes it so the method is only call it if it isn't null, if you are sure that it will always be in the queue you can do a call to First instead of FirstOrDefault and replace the ?. with a simple .

Reply:
`?.` is a C#6 feature, while OpenRA is limited to C#5.

Reply:
Then you would have to do it in 3 lines with a null check I guess if you are not certain it will never be null.